---
name: Cuprum::ExceptionHandling
slug: exception-handling
type: module
files:
- lib/cuprum/exception_handling.rb
parent_path: cuprum
short_description: Utility module for handling uncaught exceptions in commands.
instance_methods:
- name: call
  path: cuprum/exception-handling/i-call
  slug: call
  inherited: false
data_path: cuprum/exception-handling
description: |-
  This functionality can be temporarily disabled by setting the
  ENV['CUPRUM_RERAISE_EXCEPTIONS'] flag; this can be used to debug issues when
  testing commands.
metadata:
  examples:
  - name: ''
    text: |-
      class UnsafeCommand < Cuprum::Command
        private

        def process
          raise 'Something went wrong.'
        end
      end

      class SafeCommand < UnsafeCommand
        include Cuprum::ExceptionHandling
      end

      UnsafeCommand.new.call
      #=> raises a StandardError

      result = SafeCommand.new.call
      #=> a Cuprum::Result
      result.error
      #=> a Cuprum::Errors::UncaughtException error.
      result.error.message
      #=> 'uncaught exception in SafeCommand -' \
      #   ' StandardError: Something went wrong.'
version: "*"
